Source/WebCore/platform/cocoa/PublicSuffixStoreCocoa.mm

```diff
@@ -54,13 +54,15 @@ static bool isPublicSuffixCF(const String& domain)
 
 String PublicSuffixStore::platformTopPrivatelyControlledDomain(StringView host) const
 {
-    size_t separatorPosition;
-    for (unsigned labelStart = 0; (separatorPosition = host.find('.', labelStart)) != notFound; labelStart = separatorPosition + 1) {
-        if (isPublicSuffix(host.substring(separatorPosition + 1)))
-            return host.substring(labelStart).toString();
-    }
+    @autoreleasepool {
```

Comment:
I don't fully understand this PR. I don't see evidence that we're allocating ObjC objects in these loops? Are we? If not, why is the autorelease pool helping?

Reply:
Yes, Objective-C objects are allocated and autoreleased within these functions. I went through all the crashes on the latest iOS 18 builds and found 3 distinct patterns. Details are in the radar.